src/com/amazon/corretto/crypto/provider/EdGen.java

```diff
@@ -35,10 +40,11 @@ public void initialize(final int keysize, final SecureRandom random) {
 
   @Override
   public KeyPair generateKeyPair() {
-    final EvpEdPrivateKey privateKey;
-    final EvpEdPublicKey publicKey;
-    privateKey = new EvpEdPrivateKey(generateEvpEdKey());
-    publicKey = privateKey.getPublicKey();
+    final EvpEdPrivateKey privateKey = new EvpEdPrivateKey(generateEvpEdKey());
+    final EvpEdPublicKey publicKey = privateKey.getPublicKey();
+    if (kf == null) { // This case indicates JDK EdDSA conditions as described in the constructor
+      return new KeyPair(publicKey, privateKey);
+    }
```

Comment:
NP: The two variable initializations could be moved inside the if block:
```
    if (kf == null) { // This case indicates JDK EdDSA conditions as described in the constructor
      final EvpEdPrivateKey privateKey = new EvpEdPrivateKey(generateEvpEdKey());
      final EvpEdPublicKey publicKey = privateKey.getPublicKey();
      return new KeyPair(publicKey, privateKey);
    }
```

Reply:
They're used below on lines 49 and 50, so need to be in scope for that usage.